app/controllers/subscriptions_controller.rb

```diff
@@ -0,0 +1,24 @@
+require_relative "../models/mailchimp"
+
+class SubscriptionsController < ApplicationController
+  # Subscribe to the MailChimp newsletter.
+  def create
+    redirect_to request.referer if !current_user
+
+    error = false
+    begin
```

Comment:
I tried something slightly different here. Would you take a look and see if you hate it? 🐩